pkg/pillar/cmd/zedrouter/ipaddronunet.go

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2021 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// Network Instance underlay network Application Number Management
+
+package zedrouter
+
+import (
+	"fmt"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+)
+
+func appNumsOnUNetAllocate(ctx *zedrouterContext,
+	config *types.AppNetworkConfig) {
+	for _, ulConfig := range config.UnderlayNetworkList {
+		appID := config.UUIDandVersion.UUID
+		networkID := ulConfig.Network
+		netStatus := lookupNetworkInstanceStatus(ctx, networkID.String())
+		if netStatus != nil {
+			baseSize := netStatus.DhcpRange.Size
+			appNum, err := appNumOnUNetAllocate(ctx, networkID, appID,
+				baseSize, false)
+			if appNum == types.AppNumInvalid || err != nil {
```

Comment:
Please please remove the definition and use of AppNumInvalid. It isn't needed and it causes confusion since there is already an error returned when it is used.